writeengine/wrapper/writeengine.h

```diff
@@ -32,9 +32,9 @@
 // end
 #include <boost/lexical_cast.hpp>
 #ifdef _MSC_VER
-#include <unordered_map>
+#include <unordered_set>
 #else
-#include <tr1/unordered_map>
+#include <tr1/unordered_set>
```

Comment:
I think we can use the `std::unordered_set` here, instead of the `tr1` version.

Reply:
No, we don't. They differ in implementation so we won't be using std::unordered_set.